xclim/indices/run_length.py

```diff
@@ -293,6 +293,163 @@ def last_run(
     return out
 
 
+def run_length_with_date(
+    da: xr.DataArray, window: int, date: str = "07-01", dim: str = "time",
+):
+    """Return the length of the longest consecutive run of True values found
+    to be semi-continuous before and after a given date.
+
+    Parameters
+    ----------
+    da : xr.DataArray
+      Input N-dimensional DataArray (boolean)
+    window : int
+      Minimum duration of consecutive run to accumulate values.
+    date:
+      The date that a run must include to be considered valid.
+    dim : str
+      Dimension along which to calculate consecutive run (default: 'time').
+
+    Returns
+    -------
+    out : xr.DataArray
+      Length of longest run of True values along a given dimension inclusive of a given date.
+
+    Notes
+    -----
+    The run can include holes of False or NaN values, so long as they do not exceed the window size.
+    """
+    include_date = datetime.strptime(date, "%m-%d").timetuple().tm_yday
+
+    def _crld(rlda, rlwindow, rldoy, rldim):
+        mid_index = np.where(rlda.time.dt.dayofyear == rldoy)[0]
+        if (
+            mid_index.size == 0
+        ):  # The date is not within the group. Happens at boundaries.
+            all_nans = xr.full_like(rlda.isel(time=0), np.nan)
+            all_nans.attrs = {}
+            return all_nans
+        end = first_run(
+            (~rlda).where(rlda.time >= rlda.time[mid_index][0]),
+            window=rlwindow,
+            dim=rldim,
+        )
+        beg = first_run(rlda, window=rlwindow, dim=rldim)
+        sl = end - beg
+        sl = xr.where(
+            beg.isnull() & end.notnull(), 0, sl
+        )  # If series is never triggered
+        sl = xr.where(
+            beg.notnull() & end.isnull(), rlda.time.size - beg, sl
+        )  # If series is not ended by end of resample time frequency
+        return sl.where(sl >= 0)
+
+    return _crld(rlda=da, rlwindow=window, rldoy=include_date, rldim=dim)
+
+
+def run_end_after_date(
```

Comment:
Je ne vois pas de tests pour ces nouvelles fonctions.

Reply:
There are no explicit tests beyond tests for the functions that they contribute to. I'll open a ticket with a high priority for writing tests for the three new run_length functions.

Reply:
See #391